addons/im_livechat/models/im_livechat_channel.py

```diff
@@ -262,10 +262,15 @@ def _get_random_operator(self, lang=None, country_id=None):
         active_channels = self.env.cr.dictfetchall()
         operator = None
         # Try to match an operator with the same lang as the visitor
+        # If no operator with the same lang, try to match an operator with the preferred lang
         if lang:
             same_lang_operator_ids = self.available_operator_ids.filtered(lambda operator: operator.partner_id.lang == lang)
             if same_lang_operator_ids:
                 operator = self._get_less_active_operator(active_channels, same_lang_operator_ids)
+            else:
+                preferred_lang_operator_ids = self.available_operator_ids.filtered(lambda operator: lang in operator.livechat_langs.mapped('code'))
```

Comment:
I think it is misleading to call it the “preferred” language. When something is said to be the “preferred”, I'd expect it to be the default, which isn't the case there

Reply:
I wanted to mean "I prefer to use this language to communicate" but I agree that can be misleading. I will switch to another name